### ISSUE TITLE
Remove `worker_threads` from `Config`

### ISSUE THREAD

**Issue by mariaschett** [project-oak/oak#2870]
We want to remove the [`worker_threads`](https://github.com/project-oak/oak/blob/6d43de3088df70dcf439c1ef3b5192946e82baa9/oak_functions/oak_functions_loader_base/src/main.rs#L63).

@tiziano88 : Do we want to keep the [default 4](https://github.com/project-oak/oak/blob/6d43de3088df70dcf439c1ef3b5192946e82baa9/oak_functions/loader/src/lib.rs#L115)?

Or not set any?

**Comment by tiziano88** [comment]
Yeah I think we always want the default, I just added the explicit option because I was debugging a race condition and I thought it was due to this, but in the end it wasn't, and I don't think we ever used it for anything else, so it should be safe to delete it.

**Comment by mariaschett** [comment]
One thing: ["Note that the CPU core detection logic does not seem to work reliably on Google Cloud Run"](https://github.com/project-oak/oak/blob/1c871cad199f62d0252ecc03cf083f1aa78b1635/oak_functions/oak_functions_loader_base/src/main.rs#L58). Is that still relevant for us?

**Comment by tiziano88** [comment]
I think that's what I suspected at the time, but it turns out that was not the case. Let's go ahead and remove it, but if we notice something weird perhaps on cloud run (which we don't even use at the moment), we could look into this again.

**Comment by mariaschett** [comment]
Sounds good, thank you!
